Add limits to number of jetty threads

spacecowboy opened the pull request:
Also updates the relevant setting and docstring. The actual maximum
value is platform (and memory) dependent, but the current limit
indicates when our code will overflow so is a hard limit.

We should refrain from specifying a lower max value IMO since someone
might want to run it on a huge machine capable of many many threads.

To find out your machine's limit, you can run this script:

`groovysh maxthreads.groovy`

``` groovy
count = 0;
while(true){
  count++;
  println(count);
  new Thread(
    new Runnable(){
      public void run() {
        try {
          Thread.sleep(10000000);
        } catch(InterruptedException e) { }
      }
    }).start();
}
```

You can also mess with the limit by tweaking your limits. See what
`ulimit -u` reports. Then set some other value: `ulimit -u XXX` and the
max thread number should have changed.

spacecowboy committed:
Add limits to number of jetty threads

Also updates the relevant setting and docstring. The actual maximum
value is platform (and memory) dependent, but the current limit
indicates when our code will overflow so is a hard limit.

We should refrain from specifying a lower max value IMO since someone
might want to run it on a huge machine capable of many many threads.

To find out your machine's limit, you can run this script:

```groovysh maxthreads.groovy```

```
count = 0;
while(true){
  count++;
  println(count);
  new Thread(
    new Runnable(){
      public void run() {
        try {
          Thread.sleep(10000000);
        } catch(InterruptedException e) { }
      }
    }).start();
}
```

You can also mess with the limit by tweaking your limits. See what
`ulimit -u` reports. Then set some other value: `ulimit -u XXX` and the
max thread number should have changed.

lutovich commented:
@spacecowboy looks good to me. test failures seem relevant because `TestJetty9WebServer` never calls `Jetty9WebServer#setMaxThreads(int)`.

spacecowboy committed:
Set a valid default value for JettyMaxThreads

Will be overwritten using the Config at runtime but it is the minimal
change to make TestJetty9WebServer green, and is just a sensible thing
to do.

spacecowboy commented:
@lutovich I made the minimal change which should make them green again.
